drivers/storage/portworx/portworx.go

```diff
@@ -432,6 +432,19 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 
 	if preflight.IsEKS() {
 		toUpdate.Annotations[pxutil.AnnotationIsEKS] = "true"
+	} else if preflight.IsGKE() {
+		toUpdate.Annotations[pxutil.AnnotationIsGKE] = "true"
+	}
+
+	if len(preflight.Instance().ProviderName()) > 0 {
+		if toUpdate.Spec.CloudStorage == nil {
+			toUpdate.Spec.CloudStorage = &corev1.CloudStorageSpec{}
+		}
+		// Cloud provider exists, if not set then set it
+		if toUpdate.Spec.CloudStorage.Provider == nil || len(*toUpdate.Spec.CloudStorage.Provider) == 0 {
+			cloudProvider := preflight.Instance().ProviderName()
+			toUpdate.Spec.CloudStorage.Provider = &cloudProvider
+		}
```

Comment:
Do we need this to be set in the storage cluster spec? We already send the cloud provider when we create the portworx pod spec. We use the same `isXXX` annotations to do that - https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/deployment.go#L948

Reply:
Yeah we need to set this here.   The  function which ends up getting called https://github.com/libopenstorage/operator/blob/dd9ed40f2b159016161d1247b435260edb88f4f8/drivers/storage/portworx/deployment.go#L825 "getCloudProvider()" only uses annotation to determine the provider.     Also Harsh told me to not use Annotation and decouple that cause it will be difficult to keep track of the combinations.

Reply:
Got it. Let's move this code to https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/portworx.go#L1171.

The above code already handles a similar case for vsphere. Also, we don't want to set the cloud provider if the user is not even using cloud drives.

Reply:
Ok done...